engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -750,6 +752,11 @@ private void execute() throws YarnException, IOException
         expiryTime = StramUserLogin.refreshTokens(tokenLifeTime, FileUtils.getTempDirectoryPath(), applicationId, conf, hdfsKeyTabFile, credentials, rmAddress, true);
       }
 
+      if (System.currentTimeMillis() > nodeReportUpdateTime) {
```

Comment:
Still see System.currentTimeMillis repeated within the loop. Please make a final variable out of it.